packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -57,63 +58,79 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
     );
   }
 
-  // Do not access this field directly; use [_httpClient] instead.
-  // We set `autoUncompress` to false to ensure that we can trust the value of
-  // the `Content-Length` HTTP header. We automatically uncompress the content
-  // in our call to [consolidateHttpClientResponseBytes].
-  static final HttpClient _sharedHttpClient = HttpClient()..autoUncompress = false;
+  static Future<Isolate> _pendingLoader;
```

Comment:
Are we ok with this isolate living for the duration of the app?  Would we ever want to kill the isolate and spawn a new one on demand?

Reply:
The trade off here is the memory overhead of the isolate and the overhead cost of starting a new isolate.
Consider a scenario were an app rapidly loads images in succession, killing and spawning a new one every time would add to the overhead.
We could consider killing the isolate if it is idle for a long duration.

Reply:
What is the overhead of adding an extra isolate?

Reply:
In debug mode the overhead is both in memory and time for code jitting. In regards to memory isolate gets its own code and heap which totals to ~20mb for trivial isolate.
In release mode the overhead is mostly in memory as spawned isolate reuses existing AOT code. Memory impact (for vm heap) totals to ~7mb for trivial isolate.

Reply:
Keeping 7mb around unable to be freed is something we should avoid if we can.  If we think the performance overhead of spawning the isolate is negligible in release mode, then it sounds like we should just spawn these on demand (does that also simplify the code if we don't need to worry about sharing the isolate?).  Otherwise, killing the isolate after an idle period sgtm.

Reply:
Spawning isolate in release mode is rather fast, but in debug mode it is on the order of several hundred milliseconds and you start with slow unoptimized code. Doing this for every image request sounds expensive. So I added a idle timer that exits worker isolate (after one minute of inactivity).

Reply:
Single shared isolate that exits after one minute of inactivity is how it is currently implemented, yes.